frontend/assets/javascripts/src/utils/user.js

```diff
@@ -82,8 +82,13 @@ define([
         };
     })();
 
+    var hasTier = function(memberDetail) {
+        return !!(memberDetail && memberDetail.tier);
+    };
```

Comment:
Yep, that was my thinking behind adding it, little more concise than doing this same check everywhere.